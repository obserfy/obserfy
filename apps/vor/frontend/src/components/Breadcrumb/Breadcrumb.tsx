import { FC } from "react"
import { Flex, ThemeUIStyleObject } from "theme-ui"

export interface BreadcrumbProps {
  sx?: ThemeUIStyleObject
}
export const Breadcrumb: FC<BreadcrumbProps> = ({ sx, children }) => (
  <Flex sx={{ ...sx, alignItems: "center", overflowX: "auto" }}>
    {children}
  </Flex>
)

export default Breadcrumb
