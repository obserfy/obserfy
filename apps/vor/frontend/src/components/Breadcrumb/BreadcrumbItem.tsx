import { FC } from "react"
import { Box, ThemeUIStyleObject } from "theme-ui"
import { Link } from "../Link/Link"

interface Props {
  to?: string
}
export const BreadcrumbItem: FC<Props> = ({ to, children }) => {
  const sx: ThemeUIStyleObject = {
    display: "inline-block",
    fontSize: 0,
    color: "textMediumEmphasis",
    whiteSpace: "nowrap",
    "&:last-child": {
      color: "textPrimary",
      "& > .splitter": {
        display: "none",
      },
    },
  }

  if (to) {
    return (
      <Link to={to} sx={sx} state={{ preserveScroll: true }}>
        <span sx={{ "&:hover": { textDecoration: "underline" } }}>
          {children}
        </span>
        <span className="splitter" sx={{ mx: 1 }}>
          /
        </span>
      </Link>
    )
  }

  return (
    <Box sx={sx}>
      {children}
      <span className="splitter">/</span>
    </Box>
  )
}

export default BreadcrumbItem
