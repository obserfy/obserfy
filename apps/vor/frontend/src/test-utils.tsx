/* eslint-disable import/no-extraneous-dependencies */
import { FC, ReactElement } from "react"
import { QueryClient, QueryClientProvider } from "react-query"
import { ThemeProvider, Box, Theme } from "theme-ui"
import { render, RenderOptions, RenderResult } from "@testing-library/react"
import {
  createHistory,
  createMemorySource,
  LocationProvider,
} from "@reach/router"
import { I18nProvider } from "@lingui/react"
import { i18n } from "@lingui/core"
import { en } from "make-plural/plurals"
import theme from "./gatsby-plugin-theme-ui"
// @ts-ignore
import enCatalog from "../i18n/lingui/en/messages.js"

// any used here because the catalogs are generated b lingui
// not our responsibility
i18n.loadLocaleData("en", { plurals: en })
i18n.load("en", enCatalog.messages as any)
i18n.activate("en")

const queryClient = new QueryClient()
const history = createHistory(createMemorySource("/"))

const AllProviders: FC = ({ children }) => {
  return (
    <I18nProvider i18n={i18n}>
      <QueryClientProvider client={queryClient}>
        <LocationProvider history={history}>
          <ThemeProvider theme={theme as Theme}>
            <Box sx={{ fontSize: [16, 20] }}>{children}</Box>
          </ThemeProvider>
        </LocationProvider>
      </QueryClientProvider>
    </I18nProvider>
  )
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "queries">
): RenderResult => render(ui, { wrapper: AllProviders, ...options })

// re-export everything
export * from "@testing-library/react"

// override render method
export { customRender as render }
