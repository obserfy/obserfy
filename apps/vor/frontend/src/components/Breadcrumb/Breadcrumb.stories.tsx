import { Meta, Story } from "@storybook/react"
import Breadcrumb, { BreadcrumbProps } from "./Breadcrumb"
import BreadcrumbItem from "./BreadcrumbItem"

export default {
  title: "Core/Breadcrumb",
  component: Breadcrumb,
  parameters: {
    componentSubtitle: "Just a simple Breadcrumb",
  },
} as Meta

const Template: Story<BreadcrumbProps> = () => (
  <Breadcrumb>
    <BreadcrumbItem to="/">Students</BreadcrumbItem>
    <BreadcrumbItem to="/">Joyce</BreadcrumbItem>
    <BreadcrumbItem to="/">Observations</BreadcrumbItem>
  </Breadcrumb>
)

export const Default = Template.bind({})
Default.args = {
  sx: {},
}
