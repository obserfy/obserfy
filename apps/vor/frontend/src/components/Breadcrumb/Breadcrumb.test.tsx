import { render } from "../../test-utils"
import { Default } from "./Breadcrumb.stories"

describe("Breadcrumb", () => {
  it("should render correctly", () => {
    const { container } = render(<Default />)
    expect(container).toMatchSnapshot()
  })
})
