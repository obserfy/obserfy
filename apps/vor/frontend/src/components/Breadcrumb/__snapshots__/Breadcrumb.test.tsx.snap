// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb should render correctly 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-size: 16px;
}

@media screen and (min-width: 40em) {
  .emotion-0 {
    font-size: 20px;
  }
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  overflow-x: auto;
}

.emotion-2 {
  display: inline-block;
  font-size: 13px;
  color: var(--theme-ui-colors-textMediumEmphasis);
  white-space: nowrap;
}

.emotion-2:last-child {
  color: var(--theme-ui-colors-textPrimary);
}

.emotion-2:last-child>.splitter {
  display: none;
}

.emotion-3:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-4 {
  margin-left: 4px;
  margin-right: 4px;
}

<div>
  <div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <a
        aria-current="page"
        class="emotion-2"
        href="/"
      >
        <span
          class="emotion-3"
        >
          Students
        </span>
        <span
          class="splitter emotion-4"
        >
          /
        </span>
      </a>
      <a
        aria-current="page"
        class="emotion-2"
        href="/"
      >
        <span
          class="emotion-3"
        >
          Joyce
        </span>
        <span
          class="splitter emotion-4"
        >
          /
        </span>
      </a>
      <a
        aria-current="page"
        class="emotion-2"
        href="/"
      >
        <span
          class="emotion-3"
        >
          Observations
        </span>
        <span
          class="splitter emotion-4"
        >
          /
        </span>
      </a>
    </div>
  </div>
</div>
`;
